conda_forge_tick/models/pr_info.py

```diff
@@ -0,0 +1,342 @@
+    """
+    Sometimes, this object is called `migrator_uid` or `MigrationUidTypedDict` in the code.
+    """
+
+    bot_rerun: bool | datetime
+    """
+    If the migration was rerun because the bot-rerun label was added to the PR, this is set to the timestamp of the
+    rerun. If no rerun was performed, this is False. There are some legacy PRs where is is True, indicating an absent
+    timestamp of a rerun that was performed.
+    """
+
+    migrator_name: MigratorName
+    """
+    The name of the migrator that created the PR. As opposed to `pre_pr_migrator_status` and `pre_pr_migrator_attempts`
+    below, the names of migrators appear here with spaces and not necessarily in lowercase.
+    """
+
+    migrator_version: int
+    """
+    The version of the migrator that created the PR.
+    """
+
+    version: str | None = None
+    """
+    If migrator_name is "Version", this fields contains the version that was migrated to.
+    Otherwise, this field is None.
+    The version is not always a VersionString: For example, it can be 1.0-6
+
+    This is a legacy field since version migration information is now stored in the `version_pr_info` object.
+    """
+
+    @model_validator(mode="after")
+    def check_version(self) -> Self:
+        if self.version is None and self.migrator_name == "Version":
+            raise ValueError(
+                "The version field must be set if migrator_name is 'Version'."
+            )
+        if self.version is not None and self.migrator_name != "Version":
+            raise ValueError(
+                "The version field must be None if migrator_name is not 'Version'."
+            )
+        return self
+
+    migrator_object_version: int | OnePlusToOne | None = None
+    """
+    This field is taken from the migration YAML for YAML-based migrations. In the migration YAML, this field is called
+    `migration_number`. Increasing the migrator object version is a way to force a migration to be rerun.
+    The version of the migrator object (see above) is different from this value.
+    https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/0d70e56969f0fcba4e6211cd93224abbfe3c919f/recipe/migrations/example.exyaml#L9-L11
+    Non-YAML migrators do not set this field.
+
+    Refer to OnePlusToOne for more information about the 1+ case (legacy typo fix).
+    """
+
+    name: str | None = None
+    """
+    The name of the migration executed by the migrator. This is only used for YAML-based migrations and is the filename
+    of the migration YAML file, without the .yaml or .yml extension. Otherwise, this field is missing.
+    """
+
+    branch: str | None = None
+    """
+    The branch that was migrated. Only set if not equal to "master" or "main", which seems a bit questionable.
+    """
+
+    # noinspection PyNestedDecorators
+    @field_validator("branch")
+    @classmethod
+    def check_branch(cls, value: str) -> str:
+        if value in ("master", "main"):
```

Comment:
We apply ABI migrations to branches other than main or master. Why do we restrict those names here?

Reply:
I think you misunderstood. The `pr_info` field cannot be `master` or `main`, not the other way around.

If I'm not mistaken, this refers to this code snippet:
https://github.com/regro/cf-scripts/blob/f0796ef8db6379164454cf766df74f608f37f5c9/conda_forge_tick/migrators/core.py#L377-L379

Reply:
Ah yes. That is for backwards compat from before we allowed migrations for other branches. Resolving this one and thank you!